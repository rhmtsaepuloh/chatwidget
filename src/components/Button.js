import React from "react";
import { MdArrowForward } from "react-icons/md";

const Button = React.memo(
  ({ buttonText, onClick, icon, background, buttonColor }) => {
    return (
      <div
        className='flex items-center justify-center mx-12 rounded box-border text-center cursor-pointer py-2'
        onClick={onClick}
        style={{ marginBottom: -20, background }}>
        <p className='text-base' style={{ color: buttonColor }}>
          {buttonText}
        </p>
        {icon && <MdArrowForward style={{ color: buttonColor }} className='text-white ml-2' />}
      </div>
    );
  }
);

export default Button;
