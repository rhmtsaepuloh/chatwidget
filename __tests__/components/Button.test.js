/* eslint-disable react/react-in-jsx-scope */
import React from "react";
import { shallow } from "enzyme";
import Button from "../../src/components/Button";

const onClick = jest.fn();
const buttonText = "Chat Aja";

describe("render correctly button components", () => {
  test("renders if conditional icon true", () => {
    const icon = true;
    const wrapper = shallow(
      <Button buttonText={buttonText} onClick={onClick} icon={icon} />
    );
    expect(wrapper).toMatchSnapshot();
  });
  test("renders if conditional icon false", () => {
    const icon = false;
    const wrapper = shallow(
      <Button buttonText={buttonText} onClick={onClick} icon={icon} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
