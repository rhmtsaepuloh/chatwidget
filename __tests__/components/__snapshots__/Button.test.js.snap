// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly button components renders if conditional icon false 1`] = `ShallowWrapper {}`;

exports[`render correctly button components renders if conditional icon true 1`] = `ShallowWrapper {}`;
